### ISSUE TITLE
[hg] rule 402 doesn't detect revision='default'

### ISSUE THREAD

**Issue by arthurlogilab** [ansible/ansible-lint#946]
##### Summary
```
    - name: name
      hg:
        repo: https://example.org/org/project
        dest: /src/project
        revision: 'default'
        purge: yes
```

gets detected as 

```
[402] Mercurial checkouts must contain explicit revision
playbook.yml:88
Task/Handler: name
```

When switching to `stable` instead of `default` the error disappears. 

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible 2.9.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/arthur/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.2 (default, Jul 16 2020, 14:00:26) [GCC 9.3.0]

ansible-lint --version
 4.2.0 
```

- ansible installation method: ubuntu package
- ansible-lint installation method: pip (in tox)

##### OS / ENVIRONMENT

Ubuntu focal
